cmd/crictl/image.go

```diff
@@ -641,13 +641,7 @@ func getAuth(creds, auth, username string) (*pb.AuthConfig, error) {
 		}, nil
 	}
 
-	if auth != "" {
-		return &pb.AuthConfig{
-			Auth: auth,
-		}, nil
-	}
-
-	return nil, nil
+	return &pb.AuthConfig{Auth: auth}, nil
```

Comment:
hm, this will result in a different path here: https://github.com/containerd/containerd/blob/59c8cf6ea5f4175ad512914dd5ce554942bf144f/internal/cri/server/images/image_pull.go#L103

Or serialization of empty object will result in `nil`?

Reply:
Yes this should result in an empty object.

Reply:
so will the comparison with `nil` be true for an empty object? I worried that the behavior of a pull will change as contianerd seems to take some host auth if nil was passed.

Reply:
@mikebrow do you know this part?